keps/sig-scheduling/5732-topology-aware-workload-scheduling/README.md

```diff
@@ -0,0 +1,990 @@
+label. The scheduler identifies a rack with sufficient capacity and schedules
+all pods there at once.
+
+#### Story 2: Workload using Interconnected DRA Devices
+
+As a cluster administrator, I want to schedule a workload that requires a set of
+specialized accelerators that are physically interconnected. I use a
+`DRAConstraint` targeting a specific `ResourceClaimTemplate`. The scheduler
+finds a set of DRA resources (ResourceSlice) that are co-located and binds the
+workload's pods to them.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- **Scheduling Latency:** Evaluating multiple placements involves running
+  filter/score plugins multiple times.
+
+  - **Mitigation:** Implement pre-filtering optimizations to reject infeasible
+    placements early based on aggregate resource availability.
+
+- **Complexity of Pod Group Scheduling:** Scheduling heterogeneous Pod Groups
+  can be complex.
+
+  - **Mitigation:** The initial version supports sequential processing of pods
+    within a PodGroup, avoiding complex backtracking or parallel processing
+    in the alpha release.
+
+## Design Details
+
+### Workload API Changes
+
+The Workload API (KEP-4671) will be extended to allow specifying group-level
+scheduling constraints. An optional `ScheduleConstraints` field is added to the
+`PodGroup` spec.
+
+```go
+// PodGroup (definition from KEP-4671, with additions)
+type PodGroup struct {
+    Name *string
+
+    // SchedulingConstraints defines group-level scheduling requirements,
+    // including topology and DRA colocation.
+    SchedulingConstraints *PodGroupSchedulingConstraints
+}
+
+// PodGroupSchedulingConstraints holds the scheduling constraints for the PodGroup.
+type PodGroupSchedulingConstraints struct {
+    // TopologyConstraints specifies desired topological placements for all pods
+    // within this PodGroup.
+    TopologyConstraints []TopologyConstraint
+
+    // DRAConstraints specifies constraints on how Dynamic Resources are allocated
+    // across the PodGroup.
+    DRAConstraints []DRAConstraint
+}
+
+// TopologyConstraint describes a desired topological colocation for all pods in the PodGroup.
+type TopologyConstraint struct {
+    // Level specifies the key of the node label representing the topology domain.
+    // All pods within the PodGroup must be colocated within the same domain instance.
+    // Examples: "topology.kubernetes.io/rack"
+    Level string
+}
+
+// DRAConstraint provides constraints on how specific DRA claims across the group should
+// be fulfilled.
+type DRAConstraint struct {
+    // ResourceClaimName specifies the name of a specific ResourceClaim
+    // within the PodGroup's pods that this constraint applies to.
+    ResourceClaimName *string
+
+    // ResourceClaimTemplateName specifies the name of a ResourceClaimTemplate.
+    // This applies to all ResourceClaim instances generated from this template.
+    ResourceClaimTemplateName *string
```

Comment:
How do these fields relate to the ResourceClaim references that Pods already have? What happens if the sets of claims referenced by a Workload and its Pods are different?

Reply:
I have updated the KEP and removed the support for DRA based constraints from the alpha version and for beta I have proposed to wait for the KEP-5729: DRA: ResourceClaim Support for Workloads to define required API and lifecycle for the PodGroup level ResourceClaims.